### PR TITLE
[A.Zhukevych]Tasks 41-50

### PR DESCRIPTION
**Pull request by Hollister009**

### CODE REVIEW

File: tasks/44_time-simulation/time-simulation.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  update(time) {
+    if (this.realtime > time) return 'Error';
+    const newTime = this.time + (time - this.realtime) * this.speed;
+    this.time = newTime;
+    this.realtime = time;
+  }
+}
+
+export {SimTime};
```
Comment (dverbovyi) [inline tasks/44_time-simulation/time-simulation.js]: export class SimTime

File: tasks/48_follow-the-spy/follow-the-spy.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const followTheSpy = (directions) => {
+  const firstRoute = directions[0];
+  directions.splice(0, 1);
+  let routes = firstRoute;
+  while (directions.length > 0) {
```
Comment (dverbovyi) [inline tasks/48_follow-the-spy/follow-the-spy.js]: you don't need while-loop here

Comment (Hollister009) [reply tasks/48_follow-the-spy/follow-the-spy.js]: My solution doesn't work without it. What should I change also to make it working without while loop ?

Comment (dverbovyi) [reply tasks/48_follow-the-spy/follow-the-spy.js]: you can check this condition inside for-loop and break the loop when the condition reach success

Comment (Hollister009) [reply tasks/48_follow-the-spy/follow-the-spy.js]: Trying to solve it without it, but I cannot find a solution yet.

Comment (Hollister009) [reply tasks/48_follow-the-spy/follow-the-spy.js]: I found the solution!

**Pull request merged**